Storage/StorageTests/CoreData/MigrationTests.swift

```diff
@@ -601,6 +601,34 @@ final class MigrationTests: XCTestCase {
         XCTAssertEqual(try targetContext.count(entityName: "SitePlugin"), 1)
         XCTAssertEqual(insertedPlugin, plugin)
     }
+
+    func test_migrating_from_50_to_51_removes_OrderCount_entities() throws {
+        // Arrange
+        let sourceContainer = try startPersistentContainer("Model 50")
+        let sourceContext = sourceContainer.viewContext
+
+        let orderCount = insertOrderCount(to: sourceContext)
+        let orderCountItem1 = insertOrderCountItem(slug: "processing", to: sourceContext)
+        let orderCountItem2 = insertOrderCountItem(slug: "completed", to: sourceContext)
+        orderCount.mutableSetValue(forKey: "items").add(orderCountItem1)
+        orderCount.mutableSetValue(forKey: "items").add(orderCountItem2)
+        try sourceContext.save()
+
+        XCTAssertEqual(try sourceContext.count(entityName: "OrderCount"), 1)
+        XCTAssertEqual(try sourceContext.count(entityName: "OrderCountItem"), 2)
+
+        let sourceEntitiesNames = sourceContainer.managedObjectModel.entitiesByName.keys
+        XCTAssertTrue(sourceEntitiesNames.contains("OrderCount"))
+        XCTAssertTrue(sourceEntitiesNames.contains("OrderCountItem"))
+
+        // Action
+        let targetContainer = try migrate(sourceContainer, to: "Model 51")
+        let targetEntitiesNames = targetContainer.managedObjectModel.entitiesByName.keys
+
+        // Assert
+        XCTAssertFalse(targetEntitiesNames.contains("OrderCount"))
+        XCTAssertFalse(targetEntitiesNames.contains("OrderCountItem"))
+    }
```

Comment:
If we try to count number of objects for deleted entity:

```swift
XCTAssertEqual(try targetContext.count(entityName: "OrderCount"), 0)
```

It will crash:
```
XCTAssertEqual failed: throwing "NSFetchRequest could not locate an NSEntityDescription for entity name 'OrderCount'"
```